Packs/CIRCL/Integrations/CirclCVESearch/CirclCVESearch_test.py

```diff
@@ -0,0 +1,136 @@
+
+
+test_data = [
+    ({"cve": "cve-2000-1234,CVE-2020-155555"}, ['response.json', 'empty_response.json'], 2),
+    ({"cve": "cve-2000-1234"}, ['response.json'], 1),
+]
+
+
+def test_indicator_creation():
+    """
+    Given:
+        A valid response from the server
+
+    When:
+        indicator_creation is being called.
+
+    Then:
+        return a Common.CVE indicator type.
+    """
+
+    with open(os.path.join(Path.cwd(), 'test_data', 'response.json')) as js:
+        response = json.load(js)
+
+    with open(os.path.join(Path.cwd(), 'test_data', 'indicator.json')) as js:
+        correct_indicator = json.load(js)
+
+    indicator = generate_indicator(response).to_context()
+    assert set(indicator["CVE(val.ID && val.ID == obj.ID)"]["Tags"]) == set(
+        correct_indicator["CVE(val.ID && val.ID == obj.ID)"]["Tags"])
+
+
+@pytest.mark.parametrize("cpe, expected_output, expected_relationships", [
+    (["cpe:2.3:a:vendor:product"],
+     ["Vendor", "Product", "Application"],
+     [EntityRelationship(name="targets",
+                         entity_a='CVE-2022-1111',
+                         entity_a_type="cve",
+                         entity_b='Vendor',
+                         entity_b_type="identity").to_context(),
+      EntityRelationship(name="targets",
+                         entity_a='CVE-2022-1111',
+                         entity_a_type="cve",
+                         entity_b='Product',
+                         entity_b_type="software").to_context()]),
+    (["cpe:2.3:h:a\:_vendor"],
+     ["A: vendor", "Hardware"],
+     [EntityRelationship(name="targets",
+                         entity_a='CVE-2022-1111',
+                         entity_a_type="cve",
+                         entity_b='A: vendor',
+                         entity_b_type="identity").to_context()]),
+    (["cpe:2.3:o:::"],
+     ["Operating-System"],
+     []),
+])
+def test_parse_cpe(cpe, expected_output, expected_relationships):
+    """
+    Given:
+        A CPE represented as a list of strings
+
+    When:
+        when parse_cpe is called
+
+    Then:
+        return a tuple of a list of tags (no empty strings) and a list of EntityRelationship objects.
+    """
+
+    tags, relationships = parse_cpe(cpe, 'CVE-2022-1111')
+    assert set(tags) == set(expected_output)
+    assert [relationship.to_context() for relationship in relationships] == expected_relationships
+
+
+@pytest.mark.parametrize("cve_id_arg,response_data,expected", test_data)
+def test_multiple_cve(cve_id_arg, response_data, expected, requests_mock):
+    """
+    Given:
+        a multiple or single CVE to fetch.
+
+    When:
+        cve_command is being called.
+
+    Then:
+        return a List of commandResults - each item representing a CVE.
+    """
+    cves = argToList(cve_id_arg.get('cve'))
+    for test_file, cve in zip(response_data, cves):
+        test_path_data = os.path.join(Path.cwd(), 'test_data', test_file)
+        with open(test_path_data) as js:
```

Comment:
insted of using the open in every method, you can use the following method, just paste it in the head of the file.


```suggestion
       def util_load_json(path: str):
            with io.open(path, mode='r', encoding='utf-8') as f:
                return json.loads(f.read())
```

Reply:
again, old code, but I will change it